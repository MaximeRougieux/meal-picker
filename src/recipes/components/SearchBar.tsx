import React from 'react';
import { TextInput } from 'react-native';
import debounce from 'lodash/debounce';
interface SearchBarProps {
  setKeyword: (text: string) => void;
}
const SearchBar = ({ setKeyword }: SearchBarProps) => {
  return (
    <TextInput onChangeText={debounce(setKeyword, 300)} testID="searchbar" />
  );
};

export default SearchBar;
