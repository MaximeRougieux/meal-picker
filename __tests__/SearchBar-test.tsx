import 'react-native';
import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import SearchBar from 'recipes/components/SearchBar';

test('renders correctly', async () => {
  const setKeyword = jest.fn();

  const { toJSON, queryByTestId, queryByDisplayValue } = render(
    <SearchBar setKeyword={setKeyword} />,
  );

  fireEvent.changeText(queryByTestId('searchbar')!, 'testtext');
  await waitFor(() => queryByDisplayValue('testtext'));
  expect(toJSON()).toMatchSnapshot();
});
