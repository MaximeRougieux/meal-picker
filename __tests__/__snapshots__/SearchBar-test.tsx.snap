// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  onChangeText={[Function]}
  testID="searchbar"
/>
`;
